src/ray/common/cgroup/cgroup_setup.cc

```
@@ -140,31 +185,35 @@ bool SetupCgroupsPreparation(const std::string &node_id) {
     return false;
   }
 
-  cgroup_v2_app_folder = absl::StrFormat("%s/ray_application_%s", cgroup_folder, node_id);
-  cgroup_v2_system_folder = absl::StrFormat("%s/ray_system_%s", cgroup_folder, node_id);
+  cgroup_v2_app_folder_ =
+      absl::StrFormat("%s/ray_application_%s", cgroup_v2_folder_, node_id);
+  cgroup_v2_system_folder_ =
+      absl::StrFormat("%s/ray_system_%s", cgroup_v2_folder_, node_id);
   const std::string cgroup_v2_app_procs =
-      ray::JoinPaths(cgroup_v2_app_folder, "cgroup.procs");
+      ray::JoinPaths(cgroup_v2_app_folder_, "cgroup.procs");
   const std::string cgroup_v2_app_subtree_control =
-      ray::JoinPaths(cgroup_v2_app_folder, "cgroup.subtree_control");
+      ray::JoinPaths(cgroup_v2_app_folder_, "cgroup.subtree_control");
   const std::string cgroup_v2_system_procs =
-      ray::JoinPaths(cgroup_v2_system_folder, "cgroup.procs");
+      ray::JoinPaths(cgroup_v2_system_folder_, "cgroup.procs");
 
   // Create the system cgroup.
-  int ret_code = mkdir(cgroup_v2_system_folder.data(), kCgroupFilePerm);
+  int ret_code = mkdir(cgroup_v2_system_folder_.data(), kCgroupFilePerm);
   if (ret_code != 0) {
     RAY_LOG(ERROR) << "Failed to create system cgroup: " << strerror(errno);
     return false;
   }
 
-  if (!MoveProcsInSystemCgroup()) {
+  // TODO(hjiang): Move GCS and raylet into system cgroup, so we need a way to know system
+  // components PID for raylet.
+  if (!MoveProcsInSystemCgroup(cgroup_v2_system_folder_)) {
     return false;
   }
   if (!EnableCgroupSubtreeControl(kRootCgroupSubtreeControl.data())) {
     return false;
```

At this point, we've created cgroups and migrated processes into the system cgroup. Should we still just return false or should we throw a fatal error b/c this should never happen.